Docs Indexing UI

RomneyDa opened the pull request:
## Description

A few main things here:
- A more stateful approach to indexing progress for docs
- Several DocsService bug fixes and (a bit too much reinventing the wheel) refactoring
- GUI elements for viewing and managing indexing status

## Checklist

- [x] The relevant docs, if any, have been updated or created

## Screenshots

Docs flow ish
![image](https://github.com/user-attachments/assets/b2289a18-8e44-4fee-9f67-7f04a78cf107)

Docs indexing statuses in "More" page
<img width="687" alt="image" src="https://github.com/user-attachments/assets/07c5164e-31ac-4f6e-ac48-8ee332abd4fb">

Indexing status peeks in chat
<img width="689" alt="image" src="https://github.com/user-attachments/assets/6230beae-992c-4a39-bf68-da13a3360e60">

<img width="737" alt="image" src="https://github.com/user-attachments/assets/ec525648-e657-4d6b-b2f3-10239d639d90">

New Add docs form flow that shows status after adding
https://www.loom.com/share/5dba457250dc4b5ab64ddaf1b853fa1f

Patrick-Erichsen reviewed (dismissed):
This is awesome, ton of nice polish, e.g. clicking on the progress from chat linking to docs, clicking on title linking to the site, the favicon, the new success dialog, the progress text updates - feels super polished 👌 

A few nitpicks/ideas:

- I have a hunch that for long running docs indexing, or indexing multiple, the footer on the homepage will become distracting, and therefore we should allow users to dismiss it. I think this could be a followup task though, not a big deal rn.
- Grabbing lastUpdated from the SQL table could be nice info for the "More" page view of docs
- When reopening the dialog after indexing a doc, it opened to the "Docs added" success page, it should probably default to going back to the "Add a new doc" page.
- The title element for codebase indexing ins't clickable, but this one is, so I think we need to make that clear through the UI. Probably either an underline by default, or a `arrow-top-right-on-square` from hero icons... but the "click to re-index" is already underlined so maybe +1 to an icon
- I think it's more typical to have the "close" button here be on the left and either be an outlined or ghost button. Also a bit smaller in general, I don't think they need to take up the whole width (especially on wider viewports)
![Screenshot 2024-11-18 at 12 29 27 PM](https://github.com/user-attachments/assets/215fa89b-255b-4954-802b-f5095ea7265e)

RomneyDa commented:
> * I have a hunch that for long running docs indexing, or indexing multiple, the footer on the homepage will become distracting, and therefore we should allow users to dismiss it. I think this could be a followup task though, not a big deal rn.

Now hideable, will only show up again if all indexing stopped for that type
<img width="656" alt="image" src="https://github.com/user-attachments/assets/5398c882-fb55-48ab-b351-d08ca838153f">

> * When reopening the dialog after indexing a doc, it opened to the "Docs added" success page, it should probably default to going back to the "Add a new doc" page.
Fixed!

> * The title element for codebase indexing ins't clickable, but this one is, so I think we need to make that clear through the UI. Probably either an underline by default, or a `arrow-top-right-on-square` from hero icons... but the "click to re-index" is already underlined so maybe +1 to an icon
Added
<img width="141" alt="image" src="https://github.com/user-attachments/assets/a44c037f-2864-44b5-bcee-ef9f73316da6">

> * I think it's more typical to have the "close" button here be on the left and either be an outlined or ghost button. Also a bit smaller in general, I don't think they need to take up the whole width (especially on wider viewports)
Had on left as more of a "go to the next thing" placement, so updated text to "Done" and made the add more button secondary
<img width="229" alt="image" src="https://github.com/user-attachments/assets/c7a2ec30-f5da-430b-9f10-6af0520c999a">

> * Grabbing lastUpdated from the SQL table could be nice info for the "More" page view of docs
will look into this
